src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -1082,87 +1082,76 @@ boolean classes(boolean handleExceptions, boolean defineAdapters, boolean conver
-            out.println("    operator       unsigned char*() { return (unsigned char*)(operator char*)(); }");
-            out.println("    operator const          char*() { size = str.size(); return                 str.c_str(); }");
-            out.println("    operator const signed   char*() { size = str.size(); return (signed   char*)str.c_str(); }");
-            out.println("    operator const unsigned char*() { size = str.size(); return (unsigned char*)str.c_str(); }");
-            out.println("    operator wchar_t*() {");
-            out.println("        const wchar_t* data = str.data();");
-            out.println("        if (str.size() > size) {");
-            out.println("            ptr = new (std::nothrow) wchar_t[str.size()+1];");
-            out.println("            if (ptr) memset(ptr, 0, sizeof(wchar_t) * (str.size()+1));");
-            out.println("        }");
-            out.println("        if (ptr && memcmp(ptr, data, sizeof(wchar_t) * str.size()) != 0) {");
-            out.println("            memcpy(ptr, data, sizeof(wchar_t) * str.size());");
-            out.println("            if (size > str.size()) ptr[str.size()] = 0;");
-            out.println("        }");
+            out.println();
+            out.println("    template<typename P>");
+            out.println("    const P* getConstPtrImpl() {");
             out.println("        size = str.size();");
-            out.println("        owner = ptr;");
-            out.println("        return ptr;");
+            out.println("        return reinterpret_cast<const P*>(str.c_str());");
             out.println("    }");
-            out.println("    operator     unsigned   short*() { return (unsigned short*)(operator wchar_t*)(); }");
-            out.println("    operator       signed     int*() { return (  signed   int*)(operator wchar_t*)(); }");
-            out.println("    operator const        wchar_t*() { size = str.size(); return                  str.c_str(); }");
-            out.println("    operator const unsigned short*() { size = str.size(); return (unsigned short*)str.c_str(); }");
-            out.println("    operator const   signed   int*() { size = str.size(); return (  signed   int*)str.c_str(); }");
-            out.println("    operator         std::basic_string<T>&() { return str; }");
-            out.println("    operator         std::basic_string<T>*() { return ptr ? &str : 0; }");
-            out.println("    T* ptr;");
-            out.println("    typename std::basic_string<T>::size_type size;");
-            out.println("    void* owner;");
-            out.println("    std::basic_string<T> str2;");
-            out.println("    std::basic_string<T>& str;");
             out.println("};");
             out.println();
+            out.println("template<typename T> const T StringAdapter<T>::emptyStringLiteral[1] = { 0 };");
```

Comment:
Let's put that as a local variable so we don't need to allocate memory or have lines all over the place for this.

Reply:
I think it would be better to completely get rid of it and call clear() in assign() when ptr is null.

Reply:
Sounds good to me!